### PR TITLE
[SPARK-9911] [DOC] [ML] Update Userguide for Evaluator

### PR DESCRIPTION
**Pull request by MechCoder**
I added a small note about the different types of evaluator and the metrics used.

### CODE REVIEW

**Comment by MechCoder** [comment]
@mengxr 

I thought it is unnecessary to add a separate guide for evaluators and hence added a note within the existing example.

File: docs/ml-guide.md [hunk 1]
```diff
@@ -645,6 +645,13 @@ An important task in ML is *model selection*, or using data to find the best mod
 Currently, `spark.ml` supports model selection using the [`CrossValidator`](api/scala/index.html#org.apache.spark.ml.tuning.CrossValidator) class, which takes an `Estimator`, a set of `ParamMap`s, and an [`Evaluator`](api/scala/index.html#org.apache.spark.ml.Evaluator).
 `CrossValidator` begins by splitting the dataset into a set of *folds* which are used as separate training and test datasets; e.g., with `$k=3$` folds, `CrossValidator` will generate 3 (training, test) dataset pairs, each of which uses 2/3 of the data for training and 1/3 for testing.
 `CrossValidator` iterates through the set of `ParamMap`s. For each `ParamMap`, it trains the given `Estimator` and evaluates it using the given `Evaluator`.
+
+The `Evaluator` can be a [`RegressionEvaluator`](api/scala/index.html#org.apache.spark.ml.RegressionEvaluator)
+for regression problems, a [`BinaryClassificationEvaluator`](api/scala/index.html#org.apache.spark.ml.BinaryClassificationEvaluator)
+for binary data or a [`MultiClassClassificationEvaluator`](api/scala/index.html#org.apache.spark.ml.MultiClassClassificationEvaluator)
```
Comment (feynmanliang) [inline docs/ml-guide.md]: nit: Oxford comma ("...binary data, or a...")

File: docs/ml-guide.md [hunk 2]
```diff
@@ -645,6 +645,13 @@ An important task in ML is *model selection*, or using data to find the best mod
...
+for multiclass problems. The default metric used to choose the best `ParamMap` can be overriden by the setMetric
```
Comment (feynmanliang) [inline docs/ml-guide.md]: \`setMetric\`

File: docs/ml-guide.md [hunk 3]
```diff
@@ -645,6 +645,13 @@ An important task in ML is *model selection*, or using data to find the best mod
 Currently, `spark.ml` supports model selection using the [`CrossValidator`](api/scala/index.html#org.apache.spark.ml.tuning.CrossValidator) class, which takes an `Estimator`, a set of `ParamMap`s, and an [`Evaluator`](api/scala/index.html#org.apache.spark.ml.Evaluator).
 `CrossValidator` begins by splitting the dataset into a set of *folds* which are used as separate training and test datasets; e.g., with `$k=3$` folds, `CrossValidator` will generate 3 (training, test) dataset pairs, each of which uses 2/3 of the data for training and 1/3 for testing.
```
Comment (feynmanliang) [inline docs/ml-guide.md]: Not related to this PR, but the MathJax formatting will override the \`code\` backticks; we should remove the backticks here.

File: docs/ml-guide.md [hunk 4]
```diff
@@ -710,9 +717,12 @@ val pipeline = new Pipeline()
 // We now treat the Pipeline as an Estimator, wrapping it in a CrossValidator instance.
 // This will allow us to jointly choose parameters for all Pipeline stages.
 // A CrossValidator requires an Estimator, a set of Estimator ParamMaps, and an Evaluator.
+// Note that the evaluator here is a BinaryClassificationEvaluator and the default metric
```
Comment (feynmanliang) [inline docs/ml-guide.md]: nit: "default metric used is areaUnderROC" -> "default areaUnderROC metric is used"

File: docs/ml-guide.md [hunk 5]
```diff
@@ -832,9 +842,12 @@ Pipeline pipeline = new Pipeline()
...
```
Comment (feynmanliang) [inline docs/ml-guide.md]: Same as L720

**Comment by feynmanliang** [comment]
+1 for @mengxr 's suggestion on JIRA to break this example into a separate `ml-evaluation.md` section; the examples in `ml-guide` are becoming monolithic and breaking another section will allow us to better document the features available in evaluators (e.g. all the metrics supported, whether to maximize or minimize the metric a la #8290)

**Comment by MechCoder** [comment]
That sounds like a great idea. But do you have better suggestions for an example? I would have to use a cross-validator and tune it anyway and I'm afraid there will be code repetition.

**Comment by feynmanliang** [comment]
I don't think we need code examples for every possible combination, otherwise combinatorial complexity is going to bite us hard.

We can:
 * List all the evaluators and their features
 * List the (currently two?) validators, CV and TrainTestSplit, and their features
 * Provide the current code example going through with the binary classification metric

**Comment by MechCoder** [comment]
> otherwise combinatorial complexity is going to bite us hard.

lol. That is not what I had meant. Sorry I should be more clear from here on.

In other words I was asking if it would be sufficient to move the Model Selection Example to the new `ml-evalation.md`?

**Comment by feynmanliang** [comment]
Oh, sorry!

I think moving the example and providing some guidance about how to choose evaluators/validators is sufficient for now.

**Comment by MechCoder** [comment]
ping @jkbradley ?

**Comment by jkbradley** [comment]
+1 for "moving the example and providing some guidance about how to choose evaluators/validators"

Something simple, but separate, which we can build upon later on.

**Comment by mengxr** [comment]
@MechCoder is busy this week. I will make a PR based on this.

**Comment by mengxr** [comment]
Since all comments are minor (or beyond this PR), I'm going to merge this into master and branch-1.5 first and then send another PR to fix issues here and some others.

**Pull request closed without merging**